expat/CMakeLists.txt

```diff
@@ -395,6 +395,13 @@ if(EXPAT_SHARED_LIBS)
     if(MSVC)
         set(expat_SRCS ${expat_SRCS} lib/libexpat.def)
     endif()
+    if(WIN32)
+        # Add DLL version
+        string(REPLACE "." "," DLL_VERSION ${CMAKE_PROJECT_VERSION})
+        add_definitions(-DVER_FILEVERSION_STR=${DLL_VERSION},0)
+        add_definitions(-DVER_FILEVERSION=${DLL_VERSION},0)
```

Comment:
Three notes here:
- We could deduplicate and maybe better self-explain the ",0" prefix by making the replace call about about `${PROJECT_VERSION}.0`.
- `dd_definitions` is global. Would you mind using `target_compile_definitions` for target `expat` instead, e.g.
  `target_compile_definitions(expat PRIVATE VER_FILEVERSION_STR=${_EXPAT_DLL_VERSION})`?
  The `-D` is not needed in that variation.
  https://cmake.org/cmake/help/latest/command/target_compile_definitions.html
- Based on the use of `VER_FILEVERSION_STR` below, it seems like we could get away with resolving `VER_FILEVERSION_STR` and only have `VER_FILEVERSION`?

Reply:
The second point (`target_compile_definitions`) does not seem to work. When doing so, the compiler does not replace the defines: `C:\libexpat\expat\win32\version.rc(2): error RC2104: undefined keyword or key name: VER_FILEVERSION [C:\libexpat\build_msvc\expat.vcxproj]`

Another solution used in the [CMake code directly](https://gitlab.kitware.com/cmake/cmake/-/blob/master/Source/CMakeLists.txt#L1201) is to pass via a `configure_file` to avoid the `add_definitions`.

I can take a look (tomorrow or this week-end) if it is preferred.